postgres/datadog_checks/postgres/config.py

```diff
@@ -17,10 +17,26 @@ class PostgresConfig:
 
     def __init__(self, instance):
         self.host = instance.get('host', '')
+        if not self.host:
+            raise ConfigurationError('Specify a Postgres host to connect to.')
         self.port = instance.get('port', '')
         if self.port != '':
             self.port = int(self.port)
+        self.user = instance.get('username', '')
+        if not self.user:
+            raise ConfigurationError('Please specify a user to connect to Postgres.')
+        self.password = instance.get('password', '')
         self.dbname = instance.get('dbname', 'postgres')
+
+        self.application_name = instance.get('application_name', 'datadog-agent')
+        if PY3 and not self.application_name.isascii():
+            raise ConfigurationError("Application name can include only ASCII characters: %s", self.application_name)
+        elif PY2:
+            try:
+                self.application_name.encode('ascii')
+            except UnicodeEncodeError:
+                raise ConfigurationError("Application name can include only ASCII characters: %s", self.application_name)
```

Comment:
(Nit) For readability, consider defining a separate `isascii()` compat helper, then using it here as…?

```suggestion
        if not isascii(self.application_name):
            raise ConfigurationError("Application name can include only ASCII characters: %s", self.application_name)
```

`isascii()` could look like this:

```python
def isascii(value):
    if PY3:
        return value.isascii()

    try:
        value.encode('ascii')
    except UnicodeEncodeError:
        return False
    else:
        return True
```